datadreamer/pipelines/generate_dataset_from_scratch.py

```diff
@@ -439,12 +463,31 @@ def main():
 
         image_generator.release(empty_cuda_cache=True)
 
+        # Split image_paths into batches
+        image_batches = split_image_paths(image_paths, args.batch_size_annotation)
+
     else:
-        # Load image paths for annotation
-        for image_path in os.listdir(save_dir):
-            # Check file extension: jpg, png, jpeg
-            if image_path.lower().endswith((".jpg", ".png", ".jpeg", ".bmp", "webp")):
-                image_paths.append(os.path.join(save_dir, image_path))
+        if args.loader_plugin:
+            if "GOOGLE_APPLICATION_CREDENTIALS" in os.environ:
+                image_batches = LOADERS_REGISTRY.get(args.loader_plugin)(
+                    view="all", dataset_id=args.dataset_id
+                )
+                print(len(image_batches))
+            else:
+                raise ValueError(
+                    "GOOGLE_APPLICATION_CREDENTIALS environment variable is not set for using the loader plugin"
```

Comment:
I suppose you could have a loader plugin which does not rely on GOOGLE_APPLICATION_CREDENTIALS